activerecord/lib/active_record/attribute_assignment.rb

```diff
@@ -10,15 +10,18 @@ module AttributeAssignment
       def _assign_attributes(attributes)
         multi_parameter_attributes  = {}
         nested_parameter_attributes = {}
+        new_attributes = attributes.dup
```

Comment:
Why just no use `attributes.stringify_keys`?

```suggestion
        attributes = attributes.stringify_keys
```

Reply:
Wouldn't that loop through the attributes twice? Once for turning the keys into strings and another one in the `each` block (which is precisely what I attempted to avoid in the PR).